### PR TITLE
fix(es/minifier): Fix a bug about `eval` of name mangler

### PR DESCRIPTION
**Pull request by kdy1**
**Description:**

 - Repro: https://github.com/avitorio/swcminify-debug

**Related issue:**

 - https://github.com/vercel/next.js/discussions/30237#discussioncomment-6288339

### CODE REVIEW

**Review by kdy1 — COMMENTED** [review]
swc-bump:
 - swc_ecma_minifier

File: crates/swc_ecma_minifier/tests/fixture/next/outstatic/1/input.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+(self["webpackChunk_N_E"] = self["webpackChunk_N_E"] || []).push([
```
Comment (jridgewell) [inline crates/swc_ecma_minifier/tests/fixture/next/outstatic/1/input.js:1]: This test case is too large for me to really understand what's changed, and the diff between this and [playground](https://play.swc.rs/?version=1.3.66&code=H4sIAAAAAAAAA41VS2%2FbMAy%2B51ewQVFImKGmG4p1CbKiKzrs0h0G7JQGhmIzsVFH9vToWrT57xPlPPxIgelimY%2BP5EeJYgaL5Wz4FxeVTB5vM6ce45%2Fx3XAOU3hX9fYGszkXlTMZmw0AZh%2BvLucRvPotwMWXz5%2FGYQfAlk4lNi8VsHWZugL51ogWPsliKxZJmSLfajY8GmyRri6PIMWxU85gGm8zi2uMCPC5KrU1EcR7lcY%2FLtcYx83AQ%2B8Nxuo8scPJYC9%2BkhqsL3sU7UUA2gsQpl9hmGSYPC7K5yFMp14m7EuFk5az87bM52E5OaDIpGEM671xC4qoVmwU0S9KnWTs%2FEE8pB%2FYg3g75eecE7HImeW8mUJKsAHytSHdNDLfc%2FMjhI9AN8vdtJJE3FekZZqX7XL6kPdIJcDrhrfCF2gpkCJVQwwghMBmmlFL%2BxKg2rKsyzlA1YPt%2BMR9g0njNymVsXAXgOHUf6R5UUlddtuttgxdFlbqFdpJS091Gq%2FVQsk1To74SorSVuRLYJJ3ItVYRQTJpCOvcao%2BDhFx7aFEvBSl%2BlY4DWdn0PhlyH2%2FSWjZyJ%2Bf8U55m0m1woZ1LfD2x4Pf1ME7uuCbYmUI6I75m5VEVMENb1Pf6jdY3N2D7bnpkb4jdfSffGq0Tivvcu21PpUfLPOXyAcQaW7kosDUa57KPIWRp2BEPIyiHv07EsZwGvVUxOYRhcblmDLqVkDrnS7TsljPgSiUmbIXn26P%2BUPJ6ljraW2nB%2FWNZgP1mzPFu1nWy1C3fMKmj7QBLGjm7ZOJqDFRcCFCWfBdl05Rz04uvIopYbLSFelvpXGVG4uaUrA9KQ8AJ47FQmotQ6lkeVIJGUZIMIiFU%2FeEL2SaMuSdFDfvH6jGqTzIt6Ne%2BEn%2BvdRrn%2FThfTg2rg7X3J8ZXEpXWPL9hUvW6osWidMaPQ3Xh71%2FnZTVZSHisqIQJkzQccP4yBCkudlqVKuQXczw1g2osjmf%2FANAdMqpjAcAAA%3D%3D&config=H4sIAAAAAAAAA32UO3LjMAxAe5%2FCozrFjostcoB0OQOHFkGZWYrQEKBtjcd3D%2FSx4o0hdRIefsTvttvvqy%2Bqq%2Ff9TT7lp7OZIC%2F%2FIqE%2Bsb2KpOK%2BA6pz6Lh6e1CmAXEuMEruE6jY5gZ4MAI6%2FDkcZoMqIhKI2NtIMMvakILvn0PW2HYZiJ5kIhWXpYXE9L%2F9zDJeaM7kSX5EjGDTBjGWTEgMDWTNcY0x2o7AnG1WvAyZ2hwItRADLAzOdBk7lScXOGCSmK%2FUgXWmRgcKChlqDmfQzCSWmCWS5ynvGbGDY2masc2%2FrOFsY7GsxITr2BLJVvF6wkBsfElaCSe4UoMJzsX9bRm8ycAlp1e7LwxppSf%2FAKQC0RIl24Lmd9TwMk9r1n7TMiQvI8u9wmW%2BtVcmaKSoJgSvVHaoDGQOWjczuFLDUNlaS2fGK%2BWj4MCA9zIrimu6BK5PWtBhz9ErQPprvTZVEzDLFq7wYSE28Ie8kvUBmzVay6d1Sn17xLgRoAU%2BodtQkFYwruMsV%2BLarfOSHMhogFNVCo3g9QjIAjCaON7Ll9mQ9RCPpol4%2FDkTs8J9OcOtTc2477f7dId3M61adGUiu6W5013%2BW%2F0oPU7wknUV6PNhOUa8fwPA2OAVLAYAAA%3D%3D) doesn't really help

**Review by jridgewell — CHANGES_REQUESTED** [review]

File: crates/swc_ecma_minifier/tests/fixture/next/outstatic/1/input.js (hunk 1 above)
Comment (kdy1) [reply crates/swc_ecma_minifier/tests/fixture/next/outstatic/1/input.js:1]: I added comments for this

Comment (kdy1) [reply crates/swc_ecma_minifier/tests/fixture/next/outstatic/1/input.js:1]: To be honest, I don't know the exact cause of the problem.

**Review by jridgewell — DISMISSED** [review]

**Pull request merged**